src/controllers/controlpickermenu.cpp

```diff
@@ -817,18 +814,21 @@ void ControlPickerMenu::addControl(QString group, QString control, QString title
                                    QString description,
                                    QMenu* pMenu,
                                    bool addReset) {
-    QAction* pAction = pMenu->addAction(title, &m_actionMapper,
-                                        SLOT(map()));
-    m_actionMapper.setMapping(pAction, m_controlsAvailable.size());
+    auto pAction = std::make_unique<QAction>(title, pMenu);
+    connect(pAction.get(), &QAction::triggered,
+            this, [this] { controlChosen(m_controlsAvailable.size()); });
```

Comment:
I think the action calls the lambda which is: 
```
slotTriggered(QAction *) {
    controlChosen(m_controlsAvailable.size());
}
```
so you will probably always use the last CO. 

how about 
```
int controlIndex = m_controlsAvailable.size();
...
[controlIndex] { controlChosen(controlIndex); }
```

Reply:
Is it possible to remove some of the duplicated code as well?